subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -643,7 +743,17 @@ static int tls_mbedtls_init(struct net_context *context, bool is_server)
 		MBEDTLS_SSL_TRANSPORT_STREAM :
 		MBEDTLS_SSL_TRANSPORT_DATAGRAM;
 
-	mbedtls_ssl_set_bio(&context->tls->ssl, context, tls_tx, tls_rx, NULL);
+	if (type == MBEDTLS_SSL_TRANSPORT_STREAM) {
+		mbedtls_ssl_set_bio(&context->tls->ssl, context,
```

Comment:
Just pondering here that should we have a way to disable TLS but leave DTLS enabled. This could happen if TCP is not enabled in Kconfig. Would this be a big change in the code?

Reply:
Well, we could add another switch to enable/disable TLS itself, yet the benefit would be very little. There are some functions in this module that are TLS specific (like `tls_rx`/`tlx_tx` bio functions), that could be compiled out, but that's it I suppose.

DTLS is based on TLS (I guess we could consider it an extension) and technically from mbedTLS side, most (if not all) TLS features are needed for DTLS operation.

As for the disabled TCP, in that case TLS socket creation will simply fail even at this point, no extra switches needed.

Reply:
I was thinking a case where TCP is disabled, in which case TCP specific code in various TLS libraries could be compiled out. Anyway, we can ponder this more later, no need to start optimize this kind of thing here.